2/debian-10/rootfs/opt/bitnami/scripts/libkafka.sh

```diff
@@ -304,66 +356,75 @@ kafka_configure_ssl_listener() {
 }
 
 ########################
-# Configure Kafka SASL_SSL listener
+# Configure Kafka for inter-broker communications
 # Globals:
-#   KAFKA_CERTIFICATE_PASSWORD
-#   KAFKA_CONF_DIR
-# Arguments:
 #   None
-# Returns:
-#   None
-#########################
-kafka_configure_sasl_ssl_listener() {
-    info "SASL_SSL listener detected, enabling SASL_SSL settings"
-    kafka_configure_ssl_listener
-    kafka_generate_jaas_authentication_file
-    # Set Kafka configuration
-    kafka_server_conf_set security.inter.broker.protocol SASL_SSL
-    kafka_server_conf_set sasl.mechanism.inter.broker.protocol PLAIN
-    kafka_server_conf_set sasl.enabled.mechanisms PLAIN,SCRAM-SHA-256,SCRAM-SHA-512
-    # Set producer/consumer configuration
-    kafka_producer_consumer_conf_set security.protocol SASL_SSL
-    kafka_producer_consumer_conf_set sasl.mechanism PLAIN
-}
-
-########################
-# Configure Kafka Only SSL listener
-# Globals:
-#   KAFKA_CERTIFICATE_PASSWORD
-#   KAFKA_CONF_DIR
 # Arguments:
-#   None
+#   $1 - Authentication protocol to use for the internal listener
 # Returns:
 #   None
 #########################
-kafka_configure_only_ssl_listener() {
-    info "SSL listener detected, enabling SSL settings"
-    kafka_configure_ssl_listener
-    # Set Kafka configuration
-    kafka_server_conf_set security.inter.broker.protocol SSL
-    # Set producer/consumer configuration
-    kafka_producer_consumer_conf_set security.protocol SSL
+kafka_configure_internal_communications() {
+    local -r protocol="${1:?missing environment variable protocol}"
+    local -r allowed_protocols=("PLAINTEXT" "SASL_PLAINTEXT" "SASL_SSL" "SSL")
+    info "Configuring Kafka for inter-broker communications with $protocol authentication."
+
+    if [[ " ${allowed_protocols[@]} " =~ " $protocol " ]]; then
+        kafka_server_conf_set security.inter.broker.protocol $protocol
+        if [[ "$protocol" = "PLAINTEXT" ]]; then
+            warn "Inter-broker communications are configured as PLAINTEXT. This is not safe for production environments."
+        fi
+        if [[ "$protocol" = "SASL_PLAINTEXT" ]] || [[ "$protocol" = "SASL_SSL" ]]; then
+            # TODO (juan131): Support other SASL implementations (e.g. GSSAPI)
+            # Please do not confuse SASL/PLAIN with PLAINTEXT (see https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_plain.html#sasl-plain-overview)
```

Comment:
I would remove the TODO and adding some wording improvements, i.e.:

```suggestion
            # The below lines would need to be updated to support other SASL implementations (i.e. GSSAPI)
            # IMPORTANT: Do not confuse SASL/PLAIN with PLAINTEXT
            # For more information, see: https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_plain.html#sasl-plain-overview)
```